### PR TITLE
GraphQl-217: module-catalog-graph-ql has an incorrect dependency on module-search

### PR DESCRIPTION
**Pull request by pmclain**
### Description (*)
Moves dependancy on Magento_Search from Magento_CatalogGraphQl to Magento_CatalogSearchGraphQl.

### Fixed Issues (if relevant)
1. magento/graphql-ce#217 : module-catalog-graph-ql has an incorrect dependency on module-search

### Manual testing scenarios (*)
```graphql
{
  products (
    filter: {
      category_id: {
        eq: "2"
      }
    }
  ) {
    items {
      id
      name
      sku
    }
    total_count
  }
}
```
The above query is executable without `Magento_Search`

```graphql
{
  products (
    search: "{{searchTerm}}"
    filter: {
      category_id: {
        eq: "2"
      }
    }
  ) {
    items {
      id
      name
      sku
    }
    total_count
  }
}
```
The next query requires `Magento_Search` through the `Magento_CatalogSearchGraphQl` module.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=304) <br/>All committers have signed the CLA.

File: app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php
Comment (paliarush) [inline app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php:92]: @pmclain does it sound right to you that this layer is called `Category layer`? Is there better name?

Comment (pmclain) [reply app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php:92]: I'm not a huge fan of the name but have grown complacent with it over the years. `CATALOG_LAYER_DEFAULT` would be my personal preference. Would this need to be done as an alias for BC? Something like this:
```php
class Resolver
{
    const CATALOG_LAYER_CATEGORY = 'category';
    const CATALOG_LAYER_SEARCH = 'search';
    const CATALOG_LAYER_DEFAULT = static::CATALOG_LAYER_CATEGORY;
...
```

Comment (paliarush) [reply app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php:92]: That is what I was thinking about, please change the constant along with its value.

We do not have any BC requirements for GraphQL classes in 2.3.0 since it was "developer preview". We only guarantee GraphQL schema for now.

Comment (pmclain) [reply app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php:92]: @paliarush Just to confirm, that constant was not introduced by graphql and predates the first stable m2 release https://github.com/magento/magento2/blame/2.3-develop/app/code/Magento/Catalog/Model/Layer/Resolver.php#L11 I don't think it is safe to change the value.

Comment (paliarush) [reply app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Filter.php:92]: You are right, I thought it was defined in CatalogGraphQL module. Let's keep it as is then. Thanks!

**Review by paliarush — DISMISSED** [review]

**Comment by pmclain** [comment]
@paliarush This needs attention after the merge of https://github.com/magento/graphql-ce/commit/45bdc4a8be79524e30633291a0cc411f5596b752#diff-9474cfb5bac8ff8a44d8df121bd593a3 Going to dismiss your review until I have a chance to rebase and resolve conflicts.

**Comment by pmclain** [comment]
@naydav Resolved the issues following the rebase and this is ready for another review.

**Comment by naydav** [comment]
@akaplya 
Please, look at this issue. 
Do we need to continue work on this task or it will be done in the scope of https://github.com/magento/graphql-ce/issues/566?

**Comment by naydav** [comment]
@pmclain, please, merge last mainline and verify that code is in actual state
We need to merge it before start working on https://github.com/magento/graphql-ce/issues/566
Thanks

**Comment by pmclain** [comment]
@naydav merged in mainline and resolved conflicts

**Comment by TomashKhamlai** [comment]
When I try to search
```graphql
{
  products (
    search: "jacket"
    filter: {
      category_id: {
        eq: "2"
      }
    }
  ) {
    items {
      id
      name
      sku
    }
    total_count
  }
}
```
I get 

```graphql
{
  "errors": [
    {
      "message": "Unknown argument \"search\" on field \"products\" of type \"Query\".",
      "category": "graphql",
      "locations": [
        {
          "line": 2,
          "column": 12
        }
      ]
    }
  ]
}
```

According to schema 
```graphql
products(
    search: String
    filter: ProductFilterInput
    pageSize: Int = 20
    currentPage: Int = 1
    sort: ProductSortInput
): Products
```

it should work.

**Comment by pmclain** [comment]
@TomashKhamlai I'm having trouble reproducing. I added a test https://github.com/magento/graphql-ce/pull/304/commits/cb10396e5ee65864edd81a6f8d623c0214e542f7 with the query above and triggered a build and it looked to pass there as well https://testing-service.magento-community.engineering/reports/magento/graphql-ce/pull/304/b9b1c036-749c-4d0c-bc65-f229460a5538/6708/WebApi/allure-report-ce-WebAPI-GRAPHQL/index.html#suites/bf7c4a1939aeaefb3fee5f5d63d990f2/a9dc3e886c0d4224/. Could you take a look at the test and let me know if there are additional steps required to reproduce?

**Comment by TomashKhamlai** [comment]
I found that `php bin/magento setup:upgrade` was omitted in my script for the case when I ask my script to switch directly into the production mode. Now it works.

**Comment by pmclain** [comment]
@naydav Closing this PR following our discussion in slack regarding the catalog resolver work being done for v2.4.x

**Pull request closed without merging**
